File: Rohit Singhal/Arrays and Strings/GlobalandLocalInversions.java
class Solution {
    public boolean isIdealPermutation(int[] nums) {
        if(nums.length<=2)
        {
            return true;
        }
        int max=nums[0];
        int i=0;
        int j=2;
        while(j<nums.length)
        {
            if(nums[j]<max)
            {
                return false;
            }
            i++;
            max=Math.max(max,nums[i]);
            
            j++;
        }
        return true;
        
    }
}

File: Hashwanth/Arrays & Strings/Global and Local Inversions/Global and Local Inversions.cpp
class Solution {
public:
    bool isIdealPermutation(vector<int>& nums) {
        
        int m=0;
        int n=nums.size();
        for(int i=0;i<n-2;i++) // for(int i=0;i<nums.size()-2;i++) -> does not work, as nums.size() returns usingned which may cause overflow with -2
        {
            m=max(m,nums[i]);
            if(m>nums[i+2])
                return false;
        }
        return true;
    }
};
